test-two-body.py

```diff
@@ -0,0 +1,49 @@
+from wrap_pyscf import *
+import unittest
+import two_body_scf_guess
+import numpy as np
+
+bs = "aug-cc-pvdz"
+mol = [("O", -0.167787, 1.645761, 0.108747),
+        ("H", 0.613411, 1.102620, 0.113724),
+        ("H", -0.093821, 2.209720, -0.643619),
+        ("O", 1.517569, -0.667424, -0.080674),
+        ("H", 1.989645, -1.098799, 0.612047),
+        ("H", 0.668397, -1.091798, -0.139744),
+        ("O", -1.350388, -0.964879, -0.092208),
+        ("H", -1.908991, -1.211298, 0.626207),
+        ("H", -1.263787, -0.018107, -0.055536)]
+
+H2OA = density(mol[0:3], bs)
+H2OB = density(mol[3:6], bs)
+H2OC = density(mol[6:9], bs)
+H4O2AB = density(mol[0:6], bs)
+H4O2AC = density((mol[0:3]+mol[6:9]), bs)
+H4O2BC = density(mol[3:9], bs)
+
+m,m = H2OA.shape
+n = m*2
+A = np.zeros((m*3, m*3))
+
+A[0:m, 0:m] = H4O2AB[0:m, 0:m]+H4O2AC[0:m, 0:m]-H2OA
+A[m:n, m:n] = H4O2AB[m:n, m:n]+H4O2BC[0:m, 0:m]-H2OB
+A[n:m*3, n:m*3] = H4O2AC[m:n, m:n]+H4O2BC[m:n, m:n]-H2OC
+A[0:m, m:n] = H4O2AB[0:m, m:n]
+A[m:n, 0:m] = H4O2AB[m:n, 0:m]
+A[0:m, n:m*3] = H4O2AC[0:m, m:n]
+A[n:m*3, 0:m] = H4O2AC[m:n, 0:m]
+A[m:n, n:m*3] = H4O2BC[0:m, m:n]
+A[n:m*3, m:n] = H4O2BC[m:n, 0:m]
+
+rho0 = A
+
+class TestTwoBodyGuess(unittest.TestCase):
+    def test_wtr_trimer(self):
+        correct_2b = rho0
+        monomers = [mol[0:3], mol[3:6], mol[6:9]]
+        dimers = [[(ms[0]+ms[1]), (ms[0]+ms[2])], [(ms[0]+ms[1]), (ms[1]+ms[2])], [(ms[0]+ms[2]), (ms[1]+ms[2])]]
+        tbg = two_body_guess(mol, monomers, dimers, bs)
+        self.assertTrue(np.allclose(correct_2b, tbg))
+        
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
I still get the same error when running the test this way.

Reply:
To make sure we're on the same page, you are still getting the error:

>======================================================================
>ERROR: /home/jacob/ (unittest.loader._FailedTest)
>----------------------------------------------------------------------
>AttributeError: module '__main__' has no attribute '/home/jacob/'
>
>----------------------------------------------------------------------
>Ran 1 test in 0.001s
>
>FAILED (errors=1)

Correct?
I can't replicate that error. Are you using the files you have checked in? Or do you have other files you are using?

Reply:
Google implies this may be due to running the code in a Jupyter Notebook. If that's the case, use 
`unittest.main(argv=['first-arg-is-ignored'], exit=False)`.

Reply:
@jake163 If you are doing your development in a Jupyter Notebook check the notebook in. Ultimately you should be checking in and backing up whatever you are using for the research.That's the only way we guarantee that your results are repeatable at a later time.

Reply:
> To make sure we're on the same page, you are still getting the error:
> 
> >======================================================================
> >ERROR: /home/jacob/ (unittest.loader._FailedTest)
> >----------------------------------------------------------------------
> >AttributeError: module '__main__' has no attribute '/home/jacob/'
> >
> >----------------------------------------------------------------------
> >Ran 1 test in 0.001s
> >
> >FAILED (errors=1)
> 
> Correct?
> I can't replicate that error. Are you using the files you have checked in? Or do you have other files you are using?

Yes.

Reply:
I made the change Jonathan suggested, and it indeed worked. Now I am getting an error related to the two_body_guess function not being defined. Both this test file and the file with the tbg functions are in the scf_guess repo, so it doesn't seem that my import statement would be wrong.

Reply:
For the time being I would ignore the Jupyter Notebook. Focus on getting your scripts so that `python3 test-two-body.py` runs your unit test.

Reply:
On the command line, I got the python interpreter to import the tbg function script by using two lines:
`import two_body_scf_guess`
`two_body_scf_guess.two_body_guess(mol, monomers, dimers, bs`

I am now getting an error saying that `mol` is not defined. I then defined `mol` and `bs` inside the unittest function (in addition to `mol` being defined non-locally) , and that did not work.

Reply:
I updated the PR to show exactly what I'm talking about. These are the precise files `python3 test_two_body.py` is executing, missing the unaltered scf main modules of course.

Reply:
I think you have some misconceptions about how Python scripts work, which have likely been brought on by starting with Jupyter Notebooks. In my experience you usually use Jupyter Notebooks to collect data and then visualize it. You don't typically do development in them (I'm sure you can it's just not what I'm used to). Right now you're doing development, which I've always done in stand-alone Python modules. Python modules are scripts meaning the interpreter starts at the top and reads to the bottom. It doesn't jump around like in a Jupyter Notebook. Try taking a look at [this](https://www.guru99.com/learn-python-main-function-with-examples-understand-main.html) tutorial on writing Python modules with main functions (I admittedly only skimmed it, but it looks like it covers scoping well enough that you should be able to resolve your issue).